core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -1724,6 +1751,39 @@ object Parser extends ParserInstances {
       null.asInstanceOf[A]
     }
 
+    final def stringIn1[A](radix: RadixNode, all: List[String], state: State): Unit = {
+      val startOffset = state.offset
+      var offset = state.offset
+      var tree = radix
+      var cont = offset < state.str.length
+      var lastMatch = -1
+      while (cont) {
+        val c = state.str.charAt(offset)
+        val idx = Arrays.binarySearch(tree.fsts, c)
+        if (idx >= 0) {
+          val prefix = tree.prefixes(idx)
+          // accept the prefix fo this character
+          if (state.str.startsWith(prefix, offset)) {
+            val children = tree.children(idx)
+            offset += prefix.length
+            tree = children
+            cont = offset < state.str.length
+            if(children.word) lastMatch = offset
+          } else {
+            cont = false
+          }
+        } else {
+          cont = false
+        }
+      }
+      if (lastMatch < 0) {
+        state.error = Chain.one(Expectation.OneStr(startOffset, all))
+        state.offset = startOffset
```

Comment:
by resetting to the original offset this parser *does* backtrack. Is that what you want? The documentation says it does not.

Reply:
Documentation was wrong (and contradicting itself), I removed the last the last false claim.

Reply:
We need to also update def doesBacktrack since this parser does backtrack, so there is no need to wrap it with a Backtrack node if a user wants backtracking.